Splitting preconditions

Ducasse opened the pull request:
Started to identify the kind of conditions mainly applicability

Ducasse committed:
Just cleaning....

jecisc committed:
Remove RPackageORganizer>>#hasRegistered

This method is not used anymore and is probably historical

jecisc committed:
Introduce #hasPackage:

Introduce #hasPackage: and use it in RPackageOrganizer. 

Subpart of #14224

Ducasse committed:
Introduced applicabilityPreconditions before splitting them into little chunks

Ducasse committed:
Merge pull request #14231 from jecisc/remove-hasRegistered

Remove RPackageORganizer>>#hasRegistered

Ducasse committed:
Extracting isNotGlobal, classExist

Ducasse commented:
Strange that they work locally and break the build.

Ducasse committed:
Merge pull request #14232 from jecisc/introduce-hasPackage

Introduce RPackageOrganizer#hasPackage:

balsa-sarenac commented on src/Refactoring2-Transformations/RBRemoveParameterTransformation.class.st:
```diff
@@ -65,14 +66,15 @@ RBRemoveParameterTransformation >> hasReferencesToTemporaryIn: each [
+
 	| imps |
-	imps := self model allImplementorsOf: oldSelector.
 	self getNewSelector.
```
Would it make sense to move this to `interactionCondition`?

Ducasse replied:
I do not know :)

balsa-sarenac commented on src/Refactoring2-Transformations/RBTemporaryToInstanceVariableTransformation.class.st:
```diff
@@ -100,6 +100,22 @@ RBTemporaryToInstanceVariableTransformation class >> model: aRBSmalltalk class:
+
+	^(RBCondition definesSelector: selector in: class)
+		& (RBCondition definesInstanceVariable: temporaryVariableName asString in: class) not
+		& (RBCondition withBlock: [
+			(class allSubclasses anySatisfy:
+				[ :cls | cls definesInstanceVariable: temporaryVariableName asString ])
+			ifTrue: [ self refactoringWarning:
+				('One or more subclasses of <1p> already defines an<n>instance variable with the same name. Proceed anyway?' expandMacrosWith: class name) ].
+			true ])
```
I think this can be qualified as a breaking change?

Ducasse replied:
Yes!

Ducasse replied:
In fact we cannot have a class with an instance variable with the same name it will break the system.

Ducasse replied:
So I do not understand the proceed anyway

balsa-sarenac reviewed (commented):
I was catching up with latest changes and while reading the code I left two minor comments.
`interatcionCondition` is nice flag for stuff that should go to driver.
I also like `self flag: #popUp` I'll add those as well so we know exact number of required changes.

Ducasse committed:
Merge 9ad50cc4e1fc0b2b4b570fb5daa29c44cc1e4859

Ducasse committed:
First break the loop in AddParameterTransformation.
Next step: 
- investigate other subclasses of myCondition
- check balsa remarks in IV
- remove myCondition from the transformations

Ducasse committed:
RBRemoveParameter green (another step in the goal of removing myCondition).

Ducasse committed:
Removing requestSuperClass since it is not necessary in PullUpTransformation

Ducasse committed:
removing unused myConditions in Transformation.

Ducasse committed:
Taking balsa remark into account
